lib/tarantool_server.py

```diff
@@ -872,6 +868,11 @@ def start(self, silent=True, wait=True, wait_load=True, rais=True, args=[],
         color_log(prefix_each_line(' | ', self.version()) + '\n',
                   schema='version')
 
+        # When the server is restarted, we need to reset port.
+        if not self.use_unix_sockets_iproto:
+            if self.iproto.port > 0:
+                self._iproto = 0
```

Comment:
TBH, this jugglish seems confusing for me:

```python
iproto = '127.0.0.1:0'
env['IPROTO'] = iproto
popen('tarantool')
iproto = get_iproto_port()  # '127.0.0.1:xxxx'
```

At different point the iproto variable means either an address to listen on (always with zero port) or an address to connect to (advertize address). I think the code would be simpler if we would separate those objects.

In fact, we don't need to store '127.0.0.1:0' anyhow, just place it to the env variable:

```python
env['IPROTO'] = '127.0.0.1:0'
open('tarantool')
iproto = get_iproto_address()
```

I don' insist, feel free to ignore.

Reply:
Agree, there is no sense to define `iproto` attribute until a real address is known.

Reply:
Fixed.